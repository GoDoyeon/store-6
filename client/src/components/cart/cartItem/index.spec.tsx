import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import CartItem from './index';

describe('<CartItem />', () => {
  it('should render same with snapshot', () => {
    const cartIdx = 1;
    const product = {
      idx: 1,
      thumbnail: 'test',
      title: 'test',
      price: 2,
    };
    const changeAmount = (price: number, type: string) => {};
    const removeCartItem = (cartId: number) => {};
    const { container } = render(
      <CartItem
        cartIdx={cartIdx}
        product={product}
        changeAmount={changeAmount}
        removeCartItem={removeCartItem}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
