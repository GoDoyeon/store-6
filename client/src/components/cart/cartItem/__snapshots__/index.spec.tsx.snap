// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> should render same with snapshot 1`] = `
.c0 {
  padding: 41px 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
}

.c1 {
  margin: 0;
  width: 90px;
  height: 120px;
  border-radius: 10px;
  -webkit-filter: drop-shadow(0px 4px 4px rgba(0,0,0,0.25)) drop-shadow(2px 2px 12px rgba(0,0,0,0.15));
  filter: drop-shadow(0px 4px 4px rgba(0,0,0,0.25)) drop-shadow(2px 2px 12px rgba(0,0,0,0.15));
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 220px;
  height: 105px;
  line-height: 1.4;
  overflow-y: auto;
  text-overflow: ellipsis;
  margin-left: 80px;
}

.c3 {
  width: 140px;
  margin-left: 60px;
}

.c4 {
  width: 140px;
  margin-left: 20px;
}

.c5 {
  margin-left: 12px;
  width: 80px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  color: #999999;
  cursor: pointer;
}

.c6:hover {
  color: #45c8c4;
}

.c7 {
  min-width: 40px;
  text-align: center;
}

.c8 {
  width: 10px;
  font-size: 25px;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  margin-left: 80px;
}

.c8:hover {
  -webkit-transition: -webkit-transform 300ms;
  -webkit-transition: transform 300ms;
  transition: transform 300ms;
  -webkit-transform: scale(1.08);
  -ms-transform: scale(1.08);
  transform: scale(1.08);
}

<div>
  <div
    class="c0"
  >
    <img
      class="c1"
      src="test"
    />
    <div
      class="c2"
    >
      test
    </div>
    <div
      class="c3"
    >
      2
    </div>
    <div
      class="c4"
    >
      2
    </div>
    <div
      class="c5"
    >
      <button
        class="c6"
      >
        ↑
      </button>
      <div
        class="c7"
      >
        1
      </div>
      <button
        class="c6"
      >
        ↓
      </button>
    </div>
    <div
      class="c8"
    >
      <img
        alt="cancle"
      />
    </div>
  </div>
</div>
`;
